### ISSUE TITLE
Player occasionally starts looping the same bit after seeking

### ISSUE THREAD

**Issue by BeMacized** [ryanheise/just_audio#12]
I am running issue where after seeking using the `seek` method, the player occasionally gets into a state where it continues looping the same part (of roughly half a second). This behavior remains even after pausing and resuming the stream.

I was not able to replicate this issue with the audio in the example application, but it does seem to occur when I replace the audio in the example with something different, for example [this](http://nextcloud.bemacized.net/s/jzdHCie6P8wXWmz) (.m4a, MPEG-4 AAC).

Seeking seems to work fine the first 4-5 times, but any more and it usually gets stuck again. 

This is on Android 10, both on an emulator as well as a physical device.
No errors appear to be thrown.

**Comment by ryanheise** [comment]
Just to clarify, does it happen with the example application when switching in your m4a file?

I have a theory on what is causing it. I added some code to "re-seek" when the initial seek missed the mark (which happens, since `MediaExtractor` actually has quite poor seek accuracy in some situations - a flaw which also exists in the higher level `MediaPlayer` API).

If this is what's happening, you'll see the plugin output "Too early, re-seeking" in logcat, generated by this code:

```java
                if (System.currentTimeMillis() - behindStartTime > BEHIND_LIMIT) {
                  System.out.println("Too early, re-seeking");
                  extractor.seekTo(start*1000L, MediaExtractor.SEEK_TO_CLOSEST_SYNC);
                  behindStartTime = 0;
                }
```

You can try commenting out this code and see if it helps. If it's the culprit, I'll delete that code.

**Comment by BeMacized** [comment]
> Just to clarify, does it happen with the example application when switching in your m4a file?

Yes.

>You can try commenting out this code and see if it helps. If it's the culprit, I'll delete that code.

Indeed, when looking in logcat i found those messages. Commenting out this section seems to have fixed my issue.

**Comment by rohansohonee** [comment]
Why am i not getting the logs printed out in my console????
Even I am facing similar issues with seek but my application just crashes abruptly. Not able to pin point the cause.
Also nothing is getting printed on console???
Log.d() and System.out.println from native code not printing in console at all??
Only dart print() is showing in my console. 
I think flutter 1.12 maybe the culprit. Can you tell me how you are getting the logs to work??

**Comment by ryanheise** [comment]
I noticed that too. To see the full logs, I run `adb logcat` directly from the command line. Some IDEs may have a built in logcat viewer, or at least Android Studio would have it.

**Comment by rohansohonee** [comment]
I am using VS Code.

**Comment by ryanheise** [comment]
I don't use VSCode, but I searched and found this: https://marketplace.visualstudio.com/items?itemName=abhiagr.logcat

**Comment by ryanheise** [comment]
I've just published the fix in 0.0.4.

@rohansohonee hopefully if you can see the logcat output for your issue, you can post it in a new issue.

**Comment by BeMacized** [comment]
> I've just published the fix in 0.0.4.

Thank you!
